qa/rpc-tests/mempool_limit.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python2
+# Copyright (c) 2014-2015 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+# Test mempool limiting together/eviction with the wallet
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+class MempoolLimitTest(BitcoinTestFramework):
+
+    def satoshi_round(self, amount):
+        return  Decimal(amount).quantize(Decimal('0.00000001'), rounding=ROUND_DOWN)
+
+    def __init__(self):
+        # Some pre-processing to create a bunch of OP_RETURN txouts to insert into transactions we create
```

Comment:
Yes. I think this code part is in `maxuploadtarget.py`, `prioritise_transaction.py` and `pruning.py`. Someone should refactor that once. I have original taken this from a test that @sdaftuar wrote.